src/sonic-host-services/scripts/hostcfgd

```diff
@@ -136,8 +158,17 @@ class FeatureHandler(object):
         # the next called self.update_feature_state will start it again. If it will fail
         # again the auto restart will kick-in. Another order may leave it in failed state
         # and not auto restart.
-        if self._cached_config[feature_name].auto_restart != feature.auto_restart:
+
+        dir_name = self.SYSTEMD_SERVICE_CONF_DIR.format(feature_name)
```

Comment:
The details of how auto restart is implemented is hidden in update_feature_auto_restart() method. Now, the implementation detail (changing systemd unit configuration) leaks into upper layer method.
I suggest to implement optimization in the update_featuer_auto_restart itself.

Reply:
Will change